tensorflow/python/eager/forwardprop_test.py

```diff
@@ -357,7 +364,10 @@ def testCustomGradientRecomputeGrad(self):
     def f(x):
```

Comment:
This test seems to fail for me locally because of the decorator - @test_util.assert_no_new_pyobjects_executing_eagerly. I am currently investigating it.

Reply:
I vaguely remember having problems with assertRaises and some of the memory decorators. Which objects does it say are leaked?

Reply:
AssertionError: The following objects were newly created: Counter({'cell': 30, 'function': 21, 'tuple': 18, 'list': 9})

Reply:
Yeah I wouldn't worry about it for this change. You can convert the test to checking for the error and then remove the memory decorator.

It's possible there's a failure to free references when returning an exception through the forward accumulator or through the tape. My first thought was that assertRaisesRegexp causes an issue, but we do have tests that use it with the memory checking decorator. Can you leave a `TODO(allenl): investigate why assert_no_new_pyobjects_executing_eagerly fails around this test`?

Reply:
Sounds good. Thanks for looking into it. It was troubling me quite a bit :). Will put the todo in my upcoming commit.